staging/src/k8s.io/component-base/config/testing/helpers.go

```diff
@@ -82,15 +79,26 @@ func RunTestsOnYAMLData(t *testing.T, scheme *runtime.Scheme, tests []TestCase,
 	}
 }
 
-func decodeTestData(path string, scheme *runtime.Scheme, gvk schema.GroupVersionKind, codecs serializer.CodecFactory) (runtime.Object, error) {
+func roundTrip(t *testing.T, path string, scheme *runtime.Scheme, outgvk schema.GroupVersionKind, codecs serializer.CodecFactory) runtime.Object {
```

Comment:
@mtaufen @liggitt sorry for being late. Is this roundtrip(external -> internal -> external) done correctly? PTAL :)

Reply:
Shouldn't we read the version off of `content` (first decode to external type and use one of the GVK accessors) and use that as the target version for converting, to complete the round-trip back to the same version? Or is that already implicitly true because of how things are set up in the filesystem/test cases?

This is the existing round trip test implementation in apimachinery. I guess we can't re-use that if we want to do (de)serialization as part of the test, but maybe we can structure our approach in a similar way?

https://github.com/kubernetes/kubernetes/blob/c7af1da206d3cc3c9d94a2c2b22abb624164db14/staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/roundtrip.go#L58